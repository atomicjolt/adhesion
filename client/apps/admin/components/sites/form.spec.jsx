import React from 'react';
import { shallow } from 'enzyme';
import Form from './form';

describe('site form', () => {
  let result;
  let props;
  let modalClosed;

  beforeEach(() => {
    modalClosed = false;
    props = {
      setupSite: () => { modalClosed = true; },
      closeModal: () => { modalClosed = true; },
      onChange: () => {},
      isUpdate: false,
    };
    result = shallow(<Form {...props} />);
  });

  it('renders the form', () => {
    expect(result).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('handles the Domain onclick event', () => {
    expect(modalClosed).toBeFalsy();
    result.find('.c-btn--yellow').simulate('click');
    expect(modalClosed).toBeTruthy();
  });

  describe('close modal', () => {
    it('closes', () => {
      expect(modalClosed).toBeFalsy();
      const button = result.find('.c-btn--gray--large');
      button.simulate('click');
      expect(modalClosed).toBeTruthy();
    });
  });
});
