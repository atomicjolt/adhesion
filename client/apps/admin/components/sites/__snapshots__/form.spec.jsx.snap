// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`site form matches the snapshot 1`] = `
<form>
  <div
    className="o-grid o-grid__modal-top"
  >
    <div
      className="o-grid__item u-full"
      key="url"
    >
      <Input
        className="c-input"
        inputProps={
          Object {
            "id": "site_url",
            "name": "url",
            "onChange": [Function],
            "type": "text",
            "value": "",
          }
        }
        labelText="Canvas Domain"
      />
    </div>
    <div
      className="o-grid__item u-full"
      key="oauth_key"
    >
      <Input
        className="c-input"
        inputProps={
          Object {
            "id": "site_oauth_key",
            "name": "oauth_key",
            "onChange": [Function],
            "type": "text",
            "value": "",
          }
        }
        labelText="Canvas Developer ID"
      />
    </div>
    <div
      className="o-grid__item u-full"
      key="oauth_secret"
    >
      <Input
        className="c-input"
        inputProps={
          Object {
            "id": "site_oauth_secret",
            "name": "oauth_secret",
            "onChange": [Function],
            "type": "text",
            "value": "",
          }
        }
        labelText="Canvas Developer Key"
      />
    </div>
  </div>
  <button
    className="c-btn c-btn--yellow"
    onClick={[Function]}
    type="button"
  >
    Create
     Domain
  </button>
  <button
    className="c-btn c-btn--gray--large u-m-right"
    onClick={[Function]}
    type="button"
  >
    Cancel
  </button>
</form>
`;
